File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/HR/Loans.cs
using LMS_CMS_DAL.Models.Domains.AccountingModule;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.HR
{
    public class Loans : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public DateOnly Date { get; set; }
        public DateOnly DeductionStartMonth { get; set; }
        public long Amount { get; set; }
        public int NumberOfDeduction { get; set; }
        public string? Notes { get; set; }

        [ForeignKey("Employee")]
        public long EmployeeID { get; set; }
        public Employee Employee { get; set; }

        [ForeignKey("Safe")]
        public long SafeID { get; set; }
        public Save Save { get; set; }

    }
}

File: LMS_CMS_Frontend/src/app/Models/HR/loans.ts
export class Loans {
    constructor(
        public id: number = 0,
        public date: string = '',
        public notes: string = '',
        public deductionStartMonth: number = 0,
        public deductionStartYear: number = 0,
        public deductionEndMonth: number = 0,
        public deductionEndYear: number = 0,
        public amount: number | null = null,
        public numberOfDeduction: number =1,
        public employeeID: number = 0,
        public employeeEnName: string = '',
        public employeeArName: string = '',
        public safeID: number = 0,
        public saveName: string = '',
    ) { }
}


